### ISSUE TITLE
How do you train the discriminator for splitting subgraphs?

### ISSUE THREAD

**Issue by xintao-xiang** [liufancs/IMP_GCN#3]
Hi authors,

Very highly motivated paper.

But how do you train the discriminator (i.e. W_2, W_3, b_2, b_3 in paper)? 
I did not find any clue in the paper. When I go to the codes, it seems like the loss bp does not use self.user_group_embeddings. Also, the self.user_group_embeddings are only used for selecting top_k (I'm not sure as I'm not familiar with Tensorflow, please correct me if I'm wrong). 
Could you please clarify how you update the parameters.

Thanks

**Comment by liufancs** [comment]
Hi,
Thanks for your interest in this paper!
First, I would like to clarify that the proposed IMP-GCN is an end-to-end model. As an input of the subgraph generation module, the user embedding can be updated with traning process.  
Then, note that we used an unsupervised method to classify users into different groups. Therefore, we did not optimize the result of the subgraph generation by updating the parameters (i.e. W_2, W_3, b_2, b_3), but by updating the inputs.
